File: src/eSims.Data/HumanResources/Person.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;

namespace eSims.Data.HumanResources
{
  public class Person
  {
    public Person()
    {
      Perks = new List<PersonPerk>();
    }

    [Key]
    public int Id { get; set; }

    public string Name { get; set; }
    [JsonConverter(typeof(StringEnumConverter))]
    public PersonState State { get; set; }
    [JsonConverter(typeof(StringEnumConverter))]
    public EmployeeState? EmployeeState { get; set; }
    [JsonConverter(typeof(StringEnumConverter))]
    public BreakType? CurrentBreak { get; set; }

    public string Image { get; set; }

    /// <summary>
    /// How good at coding stories
    /// </summary>
    public double Efficiency { get; set; }
    /// <summary>
    /// How good at bug fixes
    /// </summary>
    public double Investigation { get; set; }
    /// <summary>
    /// How good at documenting
    /// </summary>
    public double Administration { get; set; }
    /// <summary>
    /// How good at architecting
    /// </summary>
    public double Creativity { get; set; }

    /// <summary>
    /// Resiliency against sicknesses
    /// </summary>
    public double Vitality { get; set; }

    /// <summary>
    /// How easily can charm others
    /// </summary>
    public double Charisma { get; set; }
    /// <summary>
    /// How easily can accept others
    /// </summary>
    public double Empathy { get; set; }

    /// <summary>
    /// Daily energy
    /// </summary>
    public double Energy { get; set; }
    /// <summary>
    /// Overall happiness
    /// </summary>
    public double Happiness { get; set; }
    /// <summary>
    /// Daily annoyance
    /// </summary>
    public double Annoyance { get; set; }

    public List<PersonPerk> Perks { get; set; }

    public int? RoomId { get; set; }

    public int? TeamId { get; set; }

    public int? ProjectId { get; set; }

    public double? HireTime { get; set; }

    public double Pay { get; set; }

  }
}

File: src/eSims.Client/scripts/person/person.ts
const enum PersonState {
  NotAvailable,
  Available,
  Hired,
  Fired,
  Left
}

const enum EmployeeState {
  Idle,
  Working,
  OnRoute,
  OnBreak,
  AtHome,
  OnHoliday,
  OnSickLeave
}

const enum BreakType {
  Wc,
  Smokeing
}

const enum Perk {
  Smoker,
  Chimney,
  PartyPeople,
  PartyPooper,
  Leader,
  TeamPlayer,
  LoneWolf
}

export class Person {
  constructor() { }

  public id: number;
  public name: string;

  public characterState: PersonState;
  public employeeState: EmployeeState;
  public currentBreak: BreakType;

  /// <summary>
  /// How good at coding stories
  /// </summary>
  public efficiency: number;
  /// <summary>
  /// How good at bug fixes
  /// </summary>
  public investigation: number;
  /// <summary>
  /// How good at documenting
  /// </summary>
  public administration: number;
  /// <summary>
  /// How good at architecting
  /// </summary>
  public creativity: number;

  /// <summary>
  /// Resiliency against sicknesses
  /// </summary>
  public vitality: number;

  /// <summary>
  /// How easily can charm others
  /// </summary>
  public charisma: number;
  /// <summary>
  /// How easily can accept others
  /// </summary>
  public empathy: number;

  /// <summary>
  /// Daily energy
  /// </summary>
  public energy: number;
  /// <summary>
  /// Overall happiness
  /// </summary>
  public happiness: number;
  /// <summary>
  /// Daily annoyance
  /// </summary>
  public annoyance: number;

  public perks: Perk[];

  public hireTime: number;

  public pay: number;

}